src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -297,6 +326,22 @@ private Collection<Query> toQueries(Collection<QueryBuilder> queryBuilders, Quer
         return queries;
     }
 
+    private static void validatePaginationDepth(final int paginationDepth, final QueryShardContext queryShardContext) {
+        if (paginationDepth <= LOWER_BOUND_OF_PAGINATION_DEPTH) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "pagination_depth should be greater than %s", LOWER_BOUND_OF_PAGINATION_DEPTH)
+            );
+        }
+        // compare pagination depth with OpenSearch setting index.max_result_window
+        // see https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/index-settings/
+        int maxResultWindowIndexSetting = queryShardContext.getIndexSettings().getMaxResultWindow();
+        if (paginationDepth > maxResultWindowIndexSetting) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "pagination_depth should be less than index.max_result_window setting")
```

Comment:
message is not entirely correct, we're allowing values less or equal to index setting